loadgen/generic_executor.go

```diff
@@ -11,32 +11,33 @@ import (
 
 type GenericExecutor struct {
 	// Function to execute a single iteration of this scenario
-	Execute func(ctx context.Context, run *Run) error
+	Execute func(context.Context, *Run) error
 	// Default configuration if any.
 	DefaultConfiguration RunConfiguration
 }
 
-func (g GenericExecutor) GetDefaultConfiguration() RunConfiguration { return g.DefaultConfiguration }
+func (g *GenericExecutor) GetDefaultConfiguration() RunConfiguration {
+	return g.DefaultConfiguration
+}
 
 type genericRun struct {
 	executor *GenericExecutor
-	options  RunOptions
+	options  ScenarioInfo
```

Comment:
If you're renaming the type, please rename the field on this struct too.

Reply:
Will do